test/e2e/periodic_test.go

```
@@ -45,5 +52,95 @@ var _ = Describe("ControlPlaneMachineSet Operator", framework.Periodic(), func()
 			})
 		})
 
+		Context("and an instance is terminated on the cloud provider", func() {
+			var machine *machinev1beta1.Machine
+			var idx int
+
+			BeforeEach(func() {
+				client := testFramework.GetClient()
+				machineList := &machinev1beta1.MachineList{}
+				machineSelector := runtimeclient.MatchingLabels(framework.ControlPlaneMachineSetSelectorLabels())
+
+				By("Getting a list of all control plane machines")
+				Expect(client.List(testFramework.GetContext(), machineList, machineSelector)).To(Succeed(), "should be able to retrieve list of control plane machines")
+
+				machine = &machineList.Items[0]
+
+				By("Checking that the machine is actually in index 0")
+				machineIdx, err := helpers.MachineIndex(*machine)
+				Expect(machineIdx).To(Equal(0))
+				Expect(err).ToNot(HaveOccurred())
```

Switch these please, errors are always checked first else you risk masking the error with the failure of the assertion to equal 0